src/Abstractions/TaskFailureDetails.cs

```diff
@@ -35,12 +36,37 @@ public override string ToString()
     /// </remarks>
     /// <typeparam name="T">The type of exception to test against.</typeparam>
     /// <returns>
-    /// Returns <c>true</c> if the <see cref="ErrorType"/> value matches <typeparamref name="T"/>; <c>false</c> otherwise.
+    /// <c>true</c> if the <see cref="ErrorType"/> value matches <typeparamref name="T"/>; <c>false</c> otherwise.
     /// </returns>
     public bool IsCausedBy<T>() where T : Exception
     {
+        return this.IsCausedBy(typeof(T));
+    }
+
+    /// <returns>
+    /// <c>true</c> if the <see cref="ErrorType"/> value matches <paramref name="type"/>; <c>false</c> otherwise.
+    /// </returns>
+    /// <exception cref="ArgumentException">If <paramref name="type"/> is not an exception type.</exception>
+    /// <inheritdoc cref="IsCausedBy{T}"/>
+    public bool IsCausedBy(Type type)
+    {
+        if (!typeof(Exception).IsAssignableFrom(type))
+        {
+            throw new ArgumentException($"The type {type} is not an exception type.", nameof(type));
+        }
+
+        // This check works for core .NET exception types
```

Comment:
ultra nit: I suspect this may be more precise? .NET core / core .NET is overloaded (or at least I switched them in my head)
```suggestion
        // This check works for built-in .NET exception types
```

Reply:
I wanted to avoid "built-in" because the definition of "built-in" is arguably even less precise. The technical explanation is that only exceptions in System.Private.CoreLib are correctly resolved. Exceptions shipped by the .NET team that live in other assemblies, however, are not covered. This is why I chose to use "core" (with a lowercase "c" so as not to confuse with .NET Core....which clearly didn't work 😅 ).

Let me know how strongly you feel about this potential point of clarification.

Reply:
The new clarified comments work for me, thanks!